index.d.ts

```diff
@@ -0,0 +1,490 @@
+    | 'IN'
+    | 'ID'
+    | 'IR'
+    | 'IQ'
+    | 'IE'
+    | 'IM'
+    | 'IL'
+    | 'IT'
+    | 'JM'
+    | 'JP'
+    | 'JE'
+    | 'JO'
+    | 'KZ'
+    | 'KE'
+    | 'KI'
+    | 'XK'
+    | 'KW'
+    | 'KG'
+    | 'LA'
+    | 'LV'
+    | 'LB'
+    | 'LS'
+    | 'LR'
+    | 'LY'
+    | 'LI'
+    | 'LT'
+    | 'LU'
+    | 'MO'
+    | 'MG'
+    | 'MW'
+    | 'MY'
+    | 'MV'
+    | 'ML'
+    | 'MT'
+    | 'MH'
+    | 'MQ'
+    | 'MR'
+    | 'MU'
+    | 'YT'
+    | 'MX'
+    | 'FM'
+    | 'MD'
+    | 'MC'
+    | 'MN'
+    | 'ME'
+    | 'MS'
+    | 'MA'
+    | 'MZ'
+    | 'MM'
+    | 'NA'
+    | 'NR'
+    | 'NP'
+    | 'NL'
+    | 'NC'
+    | 'NZ'
+    | 'NI'
+    | 'NE'
+    | 'NG'
+    | 'NU'
+    | 'NF'
+    | 'KP'
+    | 'MK'
+    | 'MP'
+    | 'NO'
+    | 'OM'
+    | 'PK'
+    | 'PW'
+    | 'PS'
+    | 'PA'
+    | 'PG'
+    | 'PY'
+    | 'PE'
+    | 'PH'
+    | 'PN'
+    | 'PL'
+    | 'PT'
+    | 'PR'
+    | 'QA'
+    | 'RO'
+    | 'RU'
+    | 'RW'
+    | 'RE'
+    | 'WS'
+    | 'SM'
+    | 'SA'
+    | 'SN'
+    | 'RS'
+    | 'SC'
+    | 'SL'
+    | 'SG'
+    | 'SX'
+    | 'SK'
+    | 'SI'
+    | 'SB'
+    | 'SO'
+    | 'ZA'
+    | 'GS'
+    | 'KR'
+    | 'SS'
+    | 'ES'
+    | 'LK'
+    | 'BL'
+    | 'SH'
+    | 'KN'
+    | 'LC'
+    | 'MF'
+    | 'PM'
+    | 'VC'
+    | 'SD'
+    | 'SR'
+    | 'SJ'
+    | 'SE'
+    | 'CH'
+    | 'SY'
+    | 'ST'
+    | 'TW'
+    | 'TJ'
+    | 'TZ'
+    | 'TH'
+    | 'TL'
+    | 'TG'
+    | 'TK'
+    | 'TO'
+    | 'TT'
+    | 'TA'
+    | 'TN'
+    | 'TR'
+    | 'TM'
+    | 'TC'
+    | 'TV'
+    | 'UM'
+    | 'VI'
+    | 'UG'
+    | 'UA'
+    | 'AE'
+    | 'GB'
+    | 'US'
+    | 'UY'
+    | 'UZ'
+    | 'VU'
+    | 'VA'
+    | 'VE'
+    | 'VN'
+    | 'WF'
+    | 'EH'
+    | 'YE'
+    | 'ZM'
+    | 'ZW'
+    | 'AX';
+
+// source: https://developers.google.com/maps/faq#languagesupport
+type Vue2GeocoderLanguageCode =
+    | 'ja'
+    | 'sq'
+    | 'kn'
+    | 'am'
+    | 'kk'
+    | 'ar'
+    | 'km'
+    | 'hy'
+    | 'ko'
+    | 'az'
+    | 'ky'
+    | 'eu'
+    | 'lo'
+    | 'be'
+    | 'lv'
+    | 'bn'
+    | 'lt'
+    | 'bs'
+    | 'mk'
+    | 'bg'
+    | 'ms'
+    | 'my'
+    | 'ml'
+    | 'ca'
+    | 'mr'
+    | 'zh'
+    | 'mn'
+    | 'zh-CN'
+    | 'ne'
+    | 'zh-HK'
+    | 'no'
+    | 'zh-TW'
+    | 'pl'
+    | 'hr'
+    | 'pt'
+    | 'cs'
+    | 'pt-BR'
+    | 'da'
+    | 'pt-PT'
+    | 'nl'
+    | 'pa'
+    | 'en'
+    | 'ro'
+    | 'en-AU'
+    | 'ru'
+    | 'en-GB'
+    | 'sr'
+    | 'et'
+    | 'si'
+    | 'fa'
+    | 'sk'
+    | 'fi'
+    | 'sl'
+    | 'fil'
+    | 'es'
+    | 'fr'
+    | 'es-419'
+    | 'fr-CA'
+    | 'sw'
+    | 'gl'
+    | 'sv'
+    | 'ka'
+    | 'ta'
+    | 'de'
+    | 'te'
+    | 'el'
+    | 'th'
+    | 'gu'
+    | 'tr'
+    | 'iw'
+    | 'uk'
+    | 'hi'
+    | 'ur'
+    | 'hu'
+    | 'uz'
+    | 'is'
+    | 'vi'
+    | 'id'
+    | 'zu'
+    | 'it';
+
+// source: https://developers.google.com/maps/documentation/geocoding/overview#Types
+type AddressComponentType =
```

Comment:
> The following types are supported and returned by the geocoder in both the address type and address component type arrays:

Lets rename this to just `AddressType`, then we can use it in the results for the `types[]` key as well. We will have to also update it in the `AddressComponent` type

```suggestion
type AddressType =
```